src/pyhf/infer/intervals.py

```diff
@@ -64,12 +200,77 @@ def upperlimit(data, model, scan, level=0.05, return_results=False, **hypotest_k
     obs = tb.astensor([[r[0]] for r in results])
     exp = tb.astensor([[r[1][idx] for idx in range(5)] for r in results])
 
-    result_arrary = tb.concatenate([obs, exp], axis=1).T
+    result_array = tb.concatenate([obs, exp], axis=1).T
 
     # observed limit and the (0, +-1, +-2)sigma expected limits
-    limits = [_interp(level, result_arrary[idx][::-1], scan[::-1]) for idx in range(6)]
+    limits = [_interp(level, result_array[idx][::-1], scan[::-1]) for idx in range(6)]
     obs_limit, exp_limits = limits[0], limits[1:]
 
     if return_results:
         return obs_limit, exp_limits, (scan, results)
     return obs_limit, exp_limits
+
+
+def upperlimit(
+    data, model, scan=None, level=0.05, return_results=False, **hypotest_kwargs
+):
+    """
+    Calculate an upper limit interval ``(0, poi_up)`` for a single Parameter of
+    Interest (POI) using root-finding or a fixed scan through POI-space.
+
+    Example:
+        >>> import numpy as np
+        >>> import pyhf
+        >>> pyhf.set_backend("numpy")
+        >>> model = pyhf.simplemodels.uncorrelated_background(
+        ...     signal=[12.0, 11.0], bkg=[50.0, 52.0], bkg_uncertainty=[3.0, 7.0]
+        ... )
+        >>> observations = [51, 48]
+        >>> data = pyhf.tensorlib.astensor(observations + model.config.auxdata)
+        >>> scan = np.linspace(0, 5, 21)
+        >>> obs_limit, exp_limits, (scan, results) = pyhf.infer.intervals.upperlimit(
+        ...     data, model, scan, return_results=True
+        ... )
+        >>> obs_limit
+        array(1.01764089)
+        >>> exp_limits
+        [array(0.59576921), array(0.76169166), array(1.08504773), array(1.50170482), array(2.06654952)]
+
+    Args:
+        data (:obj:`tensor`): The observed data.
+        model (~pyhf.pdf.Model): The statistical model adhering to the schema ``model.json``.
+        scan (:obj:`iterable` or ``None``): Iterable of POI values or ``None`` to use
+         :class:`~pyhf.infer.intervals.upperlimit_auto_scan`.
+        level (:obj:`float`): The threshold value to evaluate the interpolated results at.
+        return_results (:obj:`bool`): Whether to return the per-point results.
+
+    Returns:
+        Tuple of Tensors:
+
+            - Tensor: The observed upper limit on the POI.
+            - Tensor: The expected upper limits on the POI.
+            - Tuple of Tensors: The given ``scan`` along with the
+              :class:`~pyhf.infer.hypotest` results at each test POI.
+              Only returned when ``return_results`` is ``True``.
+    """
+    if scan is not None:
+        return upperlimit_fixed_scan(
+            data, model, scan, level, return_results, **hypotest_kwargs
+        )
+    # else:
+    bounds = model.config.suggested_bounds()[
+        model.config.par_slice(model.config.poi_name).start
+    ]
+    relative_tolerance = hypotest_kwargs.pop("rtol", 1e-8)
```

Comment:
@alexander-held this is essentially the default relative tolerance now. Is this too tight in your mind?

Reply:
Yes, see this thread https://github.com/scikit-hep/pyhf/pull/1274#pullrequestreview-573310845. I would personally prefer a larger tolerance but am also fine with the resolution achieved there.

Reply:
This has now been backed off to `1e-4`.